packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -119,11 +128,38 @@ export default util.createRule<Options, MessageIds>({
       checkAndReportAssignmentSpace(node, leftNode, rightNode);
     }
 
+    /**
+     * Check if it is missing spaces between type annotations chaining
+     * @param typeAnnotation TypeAnnotations list
+     */
+    function checkForTypeAnnotationSpace(
+      typeAnnotation: TSESTree.TSIntersectionType | TSESTree.TSUnionType,
+    ): void {
+      const UNIONS = ['|', '&'];
+      const types = typeAnnotation.types;
+
+      types.forEach(type => {
+        const operator = sourceCode.getTokenBefore(type);
+
+        if (!!operator && UNIONS.includes(operator.value)) {
```

Comment:
nitpick: please use an explicit logical.

```suggestion
        if (operator != null && UNIONS.includes(operator.value)) {
```